app/models/order.rb

```diff
@@ -29,6 +30,22 @@ class Order < ActiveRecord::Base
 
   scope :active_orders, -> { where('status NOT IN (?)', INACTIVE_STATUS) }
 
+  def update_packages
+    if(detail_type == "GoodCity")
+      orders_packages.each do |orders_package|
+        orders_package.update_state_to_designated
+      end
+    end
+  end
+
+  def update_packages_quantity
+    if(state == "draft" && detail_type == "GoodCity")
```

Comment:
I would suggest moving this guard to the `after_create` hook that calls this function. That way, things are clearer and we are free to call `update_packages_quantity` at other times too.